### PR TITLE
Remove ip16 type, change it into uint128 type, use it in IP also

### PR DESCRIPTION
**Pull request by bradfitz**
The guts of this change are:

    // uint128 represents a uint128 using two uint64s.
    // Index 0 contains the high bits, index 1 contains the low.
    type uint128 [2]uint64

Then it replaces IP.hi and IP.lo with IP.addr of type uint128. Same
shape in memory, but with a name and type and lets us pass around both
parts together in one variable (as now used by IPRange.Prefix and
IPRange.Prefixes), including putting methods on them (moved from
the now-deleted ip16 type)

No change in inlining.

### CODE REVIEW

**Review by danderson — APPROVED** [review]
Nice! Can probably also add more methods on uint128 directly to do things like masking ops without needing the hi()/lo() accessors. I'll do a pass on that after this is merged.

**Comment by bradfitz** [comment]
Feel free to merge if it's blocking you playing. Otherwise I'll merge tomorrow giving others time to comment too if they wish.

**Comment by danderson** [comment]
All good, I'm doing minor golfing and cleanup passes, I can wait on this one.

**Review by mdlayher — APPROVED** [review]
LGTM! Now if only there was a native uint128!

**Comment by AlexanderYastrebov** [comment]
> No change in inlining.
```
$ git checkout 318330f177ab307858b73763bf7e8715db66a4a5~1 && go test -run=NONE -bench=BenchmarkParseIP > /tmp/old && git checkout 318330f177ab307858b73763bf7e8715db66a4a5 && go test -run=NONE -bench=BenchmarkParseIP > /tmp/new && benchcmp -changed /tmp/old /tmp/new
HEAD is now at 4ef2ea6 Add a few more alloc-free methods.
Previous HEAD position was 4ef2ea6 Add a few more alloc-free methods.
HEAD is now at 318330f Remove ip16 type, change it into uint128 type, use it in IP also
benchmark                              old ns/op     new ns/op     delta
BenchmarkParseIP/v4-8                  23.0          28.9          +25.65%
BenchmarkParseIP/v6-8                  77.8          92.7          +19.15%
BenchmarkParseIP/v6_ellipsis-8         56.3          70.1          +24.51%
BenchmarkParseIP/v6_v4-8               64.0          87.5          +36.72%
BenchmarkParseIP/v6_zone-8             145           163           +12.41%
BenchmarkParseIPPort/v4-8              42.8          54.2          +26.64%
BenchmarkParseIPPort/v6-8              99.5          127           +27.64%
BenchmarkParseIPPort/v6_ellipsis-8     79.9          98.7          +23.53%
BenchmarkParseIPPort/v6_v4-8           88.6          111           +25.28%
BenchmarkParseIPPort/v6_zone-8         204           197           -3.43%
```

**Pull request merged**
